### PR TITLE
Issue/72/bpz inform

### PR DESCRIPTION
**Pull request by sschmidt23**
Addressing issue #72, this PR adds an inform stage to train a prior for BPZ when given a set of "broad type" classifications for the training galaxies, and uses a newly added feature of [DESC_BPZ](https://github.com/LSSTDESC/DESC_BPZ) where rather than hardcoded values in a named prior, you can now supply a pickled dictionary of prior params to specify a prior.  
For now, the user must separately determine the "broad type" for each training galaxy, which is read in as a set of integers from file, there is not a great way to automate that process that I'm aware of, so any attempt to do so will be left to a future addition.

Inform is really a pretty straightforward likelihood minimizer that fits for the same parameterization described in the original Benitez (2000) BPZ paper and used in the default HDFN_gen prior.  It first fits for the p(broad type | apparent mag) for each broad type, then fits p(redshift | apparent mag, broad type) assuming a form z**a * exp(-z/zm)**a where zm = zo + km(m-mo), i.e. the peak of the redshift shifts linearly with apparent mag in a way that best matches the training data for each broad type.

Everything seems to work, prior looks like the training data from cosmoDC2 on a small test set, and the custom prior gives better metrics than the "default" that is not tuned to the data.

### CODE REVIEW

**Review by eacharles — APPROVED** [review]
Looks good.  Thanks!
